pkg/plugins/apiserver/server.go

```diff
@@ -0,0 +1,210 @@
+package apiserver
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/authtoken"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/live"
+	"github.com/grafana/grafana/pkg/setting"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/server"
+	liveDto "github.com/grafana/grafana-plugin-sdk-go/live"
+	grpcAuth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	logger = log.New("plugin_grpc_api")
+)
+
+func init() {
+	registry.RegisterServiceWithPriority(&GRPCAPIServer{}, registry.High)
+}
+
+// GRPCAPIServer ...
+type GRPCAPIServer struct {
+	Cfg             *setting.Cfg             `inject:""`
+	GrafanaLive     *live.GrafanaLive        `inject:""`
+	DatasourceCache datasources.CacheService `inject:""`
+	JWT             *authtoken.JWT           `inject:""`
```

Comment:
@malcolmholmes -- I got this much closer to compiling again, but the inject needs to be replaced with wire!

Reply:
Just did this - all in wire now.